sklearn/preprocessing/tests/test_data.py

```diff
@@ -2312,7 +2312,7 @@ def test_power_transformer_lambda_one():
     "method, lmbda",
     [
         ("box-cox", 0.1),
-        ("box-cox", 0.5),
+        ("box-cox", 0.2),
```

Comment:
I think the test is not generating valid data for a given value of `lambda`. Setting `lambda=0.2` resolves the issue because the generated data happens to be all less than -5.

Can we work around this by clipping `X` by `1/lambda` when the method is box-coox?

```python
X = rng.normal(loc=0, scale=1, size=(n_samples, 1))

if method == "box-cox":
    # For box-cox, means that lmbda * y + 1 > 0 or y > - 1 / lmbda
    # Clip the data here to make sure the inequality is valid.
    X = np.clip(X, - 1 / lmbda, None)
```

Reply:
Thank you! Clipping the data is a good idea, but may need to add a safety number, e.g., 1e-5.
```python
X = np.clip(X, - 1 / lmbda + 1e-5, None)
```

```python
from scipy.special import inv_boxcox

lmbda = 0.5
inv_boxcox(-1/lmbda, lmbda) # 0.0 (boxcox requires strictly positive data)
inv_boxcox(-1/lmbda + 1e-5, lmbda) # 2.5e-11
```

Reply:
Yup adding `1e-5` makes sense to me.